docs/proposals/image-provenance.md

```diff
@@ -0,0 +1,358 @@
+
+There will be a cache of decisions in the admission controller. 
+
+If the apiserver cannot reach the webhook backend, it will log a warning and either admit or deny the pod.
+A flag will control whether it admits or denys on failure.
+The rationale for deny is that an attacker could DoS the backend or wait for it to be down, and then sneak a
+bad pod into the system.  The rationale for allow here is that, if the cluster admin also does
+after-the-fact auditing of what images were run (which we think will be common), this will catch
+any bad images run during periods of backend failure.  With default-allow, the availability of Kubernetes does
+not depend on the availability of the backend.
+
+# Webhook Backend
+
+The admission controller code in that directory does not contain logic to make an admit/reject decision.  Instead, it extracts
+relevant fields from the Pod creation/update request and sends those fields to a Backend (which we have been loosely calling "WebHooks"
+in Kubernetes).  The request the admission controller sends to the backend is called a WebHook request to distinguish it from the
+request being admission-controlled.  The server that accepts the WebHook request from Kubernetes is called the "Backend"
+to distinguish it from the WebHook request itself, and from the API server.
+
+The whole system will work similarly to the [Authentication WebHook](
+https://github.com/kubernetes/kubernetes/pull/24902
+) or the [AuthorizationWebHook](
+https://github.com/kubernetes/kubernetes/pull/20347).
+
+The WebHook request can optionally authenticate itself to its backend using a token from a `kubeconfig` file.
+
+The WebHook request and response are JSON, and correspond to the following `go` structures:
+
+```go
+// Filename: pkg/apis/authentication.k8s.io/register.go
+package imagepolicy
+
+// ImageReview checks if the set of images in a pod are allowed.
+type ImageReview struct {
+ 	unversioned.TypeMeta
+ 
+ 	// Spec holds information about the pod being evaluated
+ 	Spec ImageReviewSpec
+ 
+ 	// Status is filled in by the backend and indicates whether the pod should be allowed.
+ 	Status ImageReviewStatus
+ }
+ 
+// ImageReviewSpec is a description of the pod creation request.
+type ImageReviewSpec struct {
+	// Containers is a list of a subset of the information in each container of the Pod being created.
+	Containers []ImageReviewContainerSpec
+	// Annotations is a list of key-value pairs extracted from the Pod's annotations.
+	// It only includes keys which match the pattern `*.image-policy.k8s.io/*`.
+	// It is up to each webhook backend to determine how to interpret these annotations, if at all.
+	Annotations map[string]string
+	// Namespace is the namespace the pod is being created in.
+	Namespace string
+}
+
+// ImageReviewContainerSpec is a description of a container within the pod creation request.
+type ImageReviewContainerSpec struct {
+	Image string
```

Comment:
Images can be specified to docker, and in `pod.spec.container[].image` as either `image:tag` or `image@SHA:012345679abcdef`.  So, this field also accepts either format.

It is up to the backend to decide if it accepts `image:tag` or only accepts `image@SHA:012345679abcdef` format.   There are reasons you might chose to do either way, so this API doesn't have an opinion.